examples/with-cookie-auth/pages/_document.js

```diff
@@ -0,0 +1,24 @@
+import Document, { Head, Main, NextScript } from 'next/document'
+import { getServerSideToken, getUserScript } from '../lib/auth'
+
+export default class extends Document {
+  static async getInitialProps (ctx) {
+    const props = await Document.getInitialProps(ctx)
+    const info = getServerSideToken(ctx.req)
+    return { ...props, ...info }
+  }
+
+  render () {
+    const { user = {} } = this.props
+    return (
+      <html lang='en'>
+        <Head />
+        <body>
+          <Main />
+          <script dangerouslySetInnerHTML={{ __html: getUserScript(user) }} />
```

Comment:
This is not needed, you can use `_app.js` and then assign to `window.__USER__` in constructor of _app.js

Reply:
Ah yes, no longer needed... This was written before _app I think... :)